src/problem-solver.cc

```diff
@@ -410,6 +421,18 @@ namespace hpp {
       }
     }
 
+    void ProblemSolver::initDistance ()
+    {
+      if (!problem_) throw std::runtime_error ("The problem is not defined.");
+      if(distanceType_ != "WeighedDistance"){   // default distance, no need to change
```

Comment:
I understand you do not want to rebuild the distance object. Am I right ?

However, this is a bit dangerous. If you set `distanceType_` of your type, call initDistance, then set it back to `"WeighedDistance"`, the distance is never rebuilt.
Does it matter for you to have the same distance object created twice ?